rfcs/0016-verification/0016-verification.md

```diff
@@ -0,0 +1,130 @@
+
+        The version field is for doing parallel soft forking deployments. Currently, block version MUST be zero.
+
+    4. timestamp verification
+
+        Timestamp (Unix’s time in milliseconds) of block timestamp MUST greater than the median timestamp of previous 11 blocks and less than the network-adjusted time + MAX_ALLOWED_FUTURE_TIME.
+
+    5. difficulty verification
+
+        We define previous epoch uncle rate u_last, previous epoch difficulty Diff_last, target uncle rate u.
+        The canonical difficulty of a block of header is defined as Diff == Max(Diff_last * u_last / u, Diff_last * 2), It MUST greater or equal genesis' difficulty.
+
+2. Block Verification
+
+   1. transaction uniqueness verification
+
+        All transactions included in block must be different.
+
+   2. cellbase
+
+      * required: Block MUST have exactly one cellbase.
+      * position: cellbase MUST be the first transaction in block.
+      * input: cellbase MUST contain one null input, the input's script fill binary with current block number.
+      * output_capacity must less or equal than block reward. Block reward includes primary issuance, secondary issuance and transaction fees, according to RFC#0015.
+
+   3. merkle_root
+
+       * txs_proposal: The Blake2b 256-bit hash of the root node of the [tree structure][2] populated with each transaction's proposal short id of the block.
+       * txs_commit: The Blake2b 256-bit hash of the root node of the [tree structure][2] populated with each transaction id of the block.
+
+        txs_proposal and txs_commit MUST be valid merkle proof[[2]].
+
+   4. transaction proposal rules
+
+        Commit transaction must be proposed within consensus rules specify proposal window.
+
+   5. uncles
+
+      * uncles hash: The hash MUST match the [CFB][3] serialized uncles.
+      * number: there are at most 2 uncles included in one block.
+      * depth: we define uncle number H(u), block number H(b), uncle number MUST satisfy H(b) - m < H(u) < H(b). Currently, m = 8.
+      * epoch: uncle MUST be the same epoch with block.
+      * uniqueness: an uncle must be different from all uncles included in previous blocks and all other uncles included in the same block (non-double-inclusion).
+
+   6. block size limit
+
+        The block size MUST be less or equal than block size limit. The current block size limit is 10_000_000 bytes (10MB).
+
+   7. cycles limit
+
+        The maximum cycles expenditure per block. The current cycles limit is 100_000_000.
+
+3. Transaction Verification
+
+    Cellbase transaction is excluded from following rules unless otherwise noted.
+
+    1. transaction version verification
+
+        The version field is for doing parallel soft forking deployments. Currently, transaction version MUST be zero.
+
+    2. null verification
+
+        All transaction inputs MUST not be null.
+
+    3. empty inputs/outputs verification
+
+        Transaction inputs or outputs MUST not empty.
```

Comment:
I'd suggest to emphasize the "not" part as well. 

Only make the "MUST" bold is really misleading on a first glance.